File: ECommerce/Domain/Purchase.cs
namespace Domain
{
    public class Purchase
    {
        public Guid Id { get; set; } = Guid.NewGuid();
        public Guid UserId { get; set; }
        public List<PurchasedProduct> Products { get; set; }
        public Promotion Promotion { get; set; }
        public DateTime Date { get; set; } = DateTime.Now;
        public PaymentType PaymentMethod { get; set; }

        public enum PaymentType
        {
            Visa,
            Mastercard,
            Santander,
            Itau,
            BBVA,
            Paypal,
            Paganza
        }
    }
}

File: frontend/src/app/Interfaces/purchase.ts
import { Product } from "./product"
import { Promotion } from "./promotion"

export enum PaymentType {
    Visa = 0,
    MasterCard = 1,
    Santander = 2,
    Itau = 3,
    BBVA = 4,
    Paypal = 5,
    Paganza = 6
}

export interface Purchase {
    id: string,
    userId: string,
    products: Product[],
    promotion: Promotion,
    date: string,
    paymentMethod: PaymentType,
    message: string
}